pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -242,17 +248,40 @@ func (configgen *ConfigGeneratorImpl) buildSidecarListeners(env *model.Environme
 				filter.ConfigType = &listener.Filter_Config{Config: util.MessageToStruct(tcpProxy)}
 			}
 
+			filterChains := []listener.FilterChain{
+				{
+					Filters: []listener.Filter{filter},
+				},
+			}
+
+			// The virtual listener will handle all traffic that does not match any other listeners, and will
+			// blackhole/passthrough depending on the outbound traffic policy. When passthrough is enabled,
+			// this has the risk of triggering infinite loops when requests are sent to the pod's IP, as it will
+			// send requests to itself. To block this we add an additional filter chain before that will always blackhole.
+			if pilot.RestrictPodIPTrafficLoops.Get() {
+				var cidrRanges []*core.CidrRange
+				for _, ip := range node.IPAddresses {
+					cidrRanges = append(cidrRanges, util.ConvertAddressToCidr(ip))
+				}
+				blackhole := blackholeStructMarshalling
```

Comment:
It might be easier to read as `blackhole := newBlackholeFilter(util.IsXDSMarshalingToAnyEnabled(node))`
I am not a go expert and you should decide

Reply:
this is precomputed, if I did it that way (which I did before) it would be created each time. This way we share the same object